diff --git a/Transition/Transition.jsx b/Transition/Transition.jsx
index cb18e410a4bee8d8cc902a9268162310ccf31b8d..2592f4a311c8ca9b3c9f86d07cc3f813762b4e88 100644
--- a/Transition/Transition.jsx
+++ b/Transition/Transition.jsx
@@ -114,14 +114,25 @@ export function Transition(inProps) {
     }));
     createEffect((firstTime) => {
         if (props.in) {
-            untrack(() => props.onEnter?.());
+        untrack(() => {
+            if (status() !== "entering" && status() !== "entered") {
+            props.onEnter?.();
             setStatus("entering");
-        }
-        else {
+            }
+        });
+        } else {
+        untrack(() => {
             if (!firstTime) {
-                untrack(() => props.onExit?.());
+            if (
+                status() !== "exiting" &&
+                status() !== "exited" &&
+                status() !== "unmounted"
+            ) {
+                props.onExit?.();
                 setStatus("exiting");
             }
+            }
+        });
         }
         return false;
     }, true);
